satpy/readers/mirs.py

```diff
@@ -0,0 +1,486 @@
+SENSOR = {"n18": amsu,
+          "n19": amsu,
+          "n20": 'atms',
+          "np": amsu,
+          "m1": amsu,
+          "m2": amsu,
+          "m3": amsu,
+          "ma1": amsu,
+          "ma2": amsu,
+          "ma3": amsu,
+          "npp": "atms",
+          "jpss": "atms",
+          "f17": "ssmis",
+          "f18": "ssmis",
+          "gpm": "GPI",
+          }
+
+
+def read_atms_limb_correction_coeffs(fn):
+    """Read provided limb correction files for atms."""
+    if os.path.isfile(fn):
+        coeff_str = open(fn, "r").readlines()
+    else:
+        parts = fn.split(":")
+        mod_part, file_part = parts if len(parts) == 2 else ("", parts[0])
+        mod_part = mod_part or __package__  # self.__module__
+        coeff_str = get_resource_string(mod_part, file_part).decode().split("\n")
+    # make it a generator
+    coeff_str = (line.strip() for line in coeff_str)
+
+    all_coeffs = np.zeros((22, 96, 22), dtype=np.float32)
+    all_amean = np.zeros((22, 96, 22), dtype=np.float32)
+    all_dmean = np.zeros(22, dtype=np.float32)
+    all_nchx = np.zeros(22, dtype=np.int32)
+    all_nchanx = np.zeros((22, 22), dtype=np.int32)
+    all_nchanx[:] = 9999
+    # There should be 22 sections
+    for chan_idx in range(22):
+        # blank line at the start of each section
+        _ = next(coeff_str)
+
+        # section header
+        nx, nchx, dmean = [x.strip() for x in next(coeff_str).split(" ") if x]
+        nx = int(nx)  # Question, was this supposed to be used somewhere?
+        all_nchx[chan_idx] = nchx = int(nchx)
+        all_dmean[chan_idx] = float(dmean)
+
+        # coeff locations (indexes to put the future coefficients in)
+        locations = [int(x.strip()) for x in next(coeff_str).split(" ") if x]
+        assert len(locations) == nchx
+        for x in range(nchx):
+            all_nchanx[chan_idx, x] = locations[x] - 1
+
+        # Read 'nchx' coefficients for each of 96 FOV (N_FOV).
+        for fov_idx in range(N_FOV):
+            # chan_num, fov_num, *coefficients, error
+            coeff_line_parts = [x.strip() for x in next(coeff_str).split(" ") if x][2:]
+            coeffs = [float(x) for x in coeff_line_parts[:nchx]]
+            ameans = [float(x) for x in coeff_line_parts[nchx:-1]]
+            # error_val = float(coeff_line_parts[-1])
+            for x in range(nchx):
+                all_coeffs[chan_idx, fov_idx, all_nchanx[chan_idx, x]] = coeffs[x]
+                all_amean[all_nchanx[chan_idx, x], fov_idx, chan_idx] = ameans[x]
+
+    return all_dmean, all_coeffs, all_amean, all_nchx, all_nchanx
+
+
+def apply_atms_limb_correction(datasets, channel_idx, dmean, coeffs, amean, nchx, nchanx):
+    """Apply the atms limb correction to the brightness temperature data."""
+    datasets = datasets.persist()
+    ds = datasets[channel_idx]
+    new_ds = np.zeros(ds.shape, dtype=ds.dtype)
+    for fov_idx in range(N_FOV):
+        new_ds[:, fov_idx] = da.map_blocks(coeff_cums_calc,
```

Comment:
I could be wrong, but doesn't this force the dask arrays to be computed (writing a dask array to a numpy array)? What if you got a list of all of the dask arrays and then concatenated them together along the proper axis?

Reply:
No.  You are correct.

Reply:
updated in f526e996

Reply:
f526e99 is wrong though.  It is the version which sends dask arrays to a function called by map_blocks.  ac4bcccb uses numpy input arrays and the function returns a numpy array.